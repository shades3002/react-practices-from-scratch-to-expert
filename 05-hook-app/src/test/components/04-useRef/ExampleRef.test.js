import React from 'react';
import { shallow } from 'enzyme';
import { ExampleRef } from '../../../components/04-useRef/ExampleRef';

describe('Test in <ExampleRef />', () => {

    const wrapper = shallow( <ExampleRef />);

    test('should show corretly', () => {
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( false );
    });

    test('should show component MultipleCustomHooks', () => {
        wrapper.find('button').simulate('click');
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( true );
    });    
    
});
