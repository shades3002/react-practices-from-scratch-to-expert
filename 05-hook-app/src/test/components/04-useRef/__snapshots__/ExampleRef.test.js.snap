// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ExampleRef /> should show corretly 1`] = `
<div>
  <h1>
    Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
